include/engine/routing_algorithms/shortest_path.hpp

```diff
@@ -114,7 +114,11 @@ class ShortestPathRouting final
                           needs_loop_forwad,
                           needs_loop_backwards);
         }
-        new_total_distance += std::min(total_distance_to_forward, total_distance_to_reverse);
+        //if no route is found between two parts of the via-route, the entire route becomes invalid
+        if( new_total_distance == INVALID_EDGE_WEIGHT )
```

Comment:
Is this a typo? Shouldn't it by `total_distance_to_forward == INVALID_EDGE_WEIGHT ||  total_distance_to_reverse == INVALID_EDGE_WEIGHT`?